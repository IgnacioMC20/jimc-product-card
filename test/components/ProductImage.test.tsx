import * as React from 'react';
import TestRenderer from 'react-test-renderer';
import { ProductImage, ProductCart } from '../../src/components';
import { product2 } from '../data/products';

describe('ProductImage', () => {

    test('debe mostrar el componente sin imagen', () => {
        const wrapper = TestRenderer.create(
            <ProductImage img='https:hola.jpg'/>
        )
        expect(wrapper.toJSON()).toMatchSnapshot()
    })

    test('debe mostrar el componente con imagen', () => {
        const wrapper = TestRenderer.create(
            <ProductCart product={product2}>
                {() => (
                    <ProductImage />
                )}
            </ProductCart>
        )
        expect(wrapper.toJSON()).toMatchSnapshot()
    })
})