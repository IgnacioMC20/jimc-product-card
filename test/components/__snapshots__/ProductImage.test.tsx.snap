// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage debe mostrar el componente con imagen 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="coffee mug"
    className="productImg undefined"
    src="./coffee-mug.png"
  />
</div>
`;

exports[`ProductImage debe mostrar el componente sin imagen 1`] = `
<img
  alt="coffee mug"
  className="productImg undefined"
  src="https:hola.jpg"
/>
`;
